components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -1758,42 +1766,41 @@ private MetaOutboundConnector createMetaOutboundConnector(ProvisioningConnectorC
+            throws IdentityProviderManagementClientException {
 
         if (federatedAuthenticatorRequest != null) {
             List<FederatedAuthenticator> federatedAuthenticators = federatedAuthenticatorRequest.getAuthenticators();
             String defaultAuthenticator = federatedAuthenticatorRequest.getDefaultAuthenticatorId();
             FederatedAuthenticatorConfig defaultAuthConfig = null;
             List<FederatedAuthenticatorConfig> fedAuthConfigs = new ArrayList<>();
             for (FederatedAuthenticator authenticator : federatedAuthenticators) {
-                FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
-                authConfig.setName(base64URLDecode(authenticator.getAuthenticatorId()));
-                authConfig.setDisplayName(getDisplayNameOfAuthenticator(authConfig.getName()));
-                authConfig.setEnabled(authenticator.getIsEnabled());
-
-                String definedByType = null;
-                if (authenticator.getDefinedBy() != null) {
-                    definedByType = authenticator.getDefinedBy().toString();
+                String authenticatorName = getDecodedAuthenticatorName(authenticator.getAuthenticatorId());
+                DefinedByType definedByType;
+                if (isNewFederatedAuthenticator) {
+                    definedByType = resolveDefinedByTypeToCreateFederatedAuthenticator(
+                            authenticator.getDefinedBy());
+                } else {
+                    definedByType = resolveDefinedByTypeToUpdateFederatedAuthenticator(authenticatorName);
                 }
-                authConfig.setDefinedByType(resolveDefinedByType(authConfig.getName(),
-                        definedByType, isNewFederatedAuthenticator));
 
-                List<org.wso2.carbon.identity.api.server.idp.v1.model.Property> authProperties =
-                        authenticator.getProperties();
-                if (IdentityApplicationConstants.Authenticator.SAML2SSO.FED_AUTH_NAME.equals(authConfig.getName())) {
-                    validateSamlMetadata(authProperties);
-                }
-                if (authProperties != null) {
-                    if (!areAllDistinct(authProperties)) {
-                        throw handleException(Response.Status.BAD_REQUEST,
-                                Constants.ErrorMessage.ERROR_CODE_INVALID_INPUT, " Duplicate properties are found in " +
-                                        "the request.");
-                    }
-                    List<Property> properties = authProperties.stream()
-                            .map(propertyToInternal)
-                            .collect(Collectors.toList());
-                    authConfig.setProperties(properties.toArray(new Property[0]));
+                if (DefinedByType.SYSTEM == definedByType) {
+                    validateAuthenticatorProperties(authenticatorName, authenticator.getProperties());
                 }
```

Comment:
Can't we move this to the factory as that is what creates system and user defined authenticator configs ?

Reply:
There are private methods in this class that are used by both validateAuthenticatorProperties and other methods. If we move them, we would either need to duplicate these private methods within the factory or relocate them to another utility class.